File: api-tester/src/main/java/com/revenuecat/apitester/java/OfferingAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.Offering;
import com.revenuecat.purchases.Package;

import java.util.List;
import java.util.Map;

@SuppressWarnings({"unused"})
final class OfferingAPI {
    static void check(final Offering offering) {
        final String identifier = offering.getIdentifier();
        final String serverDescription = offering.getServerDescription();
        final List<Package> availablePackages = offering.getAvailablePackages();

        final Package lifetime = offering.getLifetime();
        final Package annual = offering.getAnnual();
        final Package sixMonth = offering.getSixMonth();
        final Package threeMonth = offering.getThreeMonth();
        final Package twoMonth = offering.getTwoMonth();
        final Package monthly = offering.getMonthly();
        final Package weekly = offering.getWeekly();
        final Package p = offering.get("");
        final Package p2 = offering.getPackage("");

        final Map<String, Object> metadata = offering.getMetadata();
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/OfferingAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.Offering
import com.revenuecat.purchases.Package
import com.revenuecat.purchases.paywalls.PaywallData

@Suppress("unused", "UNUSED_VARIABLE")
private class OfferingAPI {
    fun check(offering: Offering) {
        with(offering) {
            val identifier: String = identifier
            val serverDescription: String = serverDescription
            val availablePackages: List<Package> = availablePackages
            val lifetime: Package? = lifetime
            val annual: Package? = annual
            val sixMonth: Package? = sixMonth
            val threeMonth: Package? = threeMonth
            val twoMonth: Package? = twoMonth
            val monthly: Package? = monthly
            val weekly: Package? = weekly
            val p1: Package = offering[""]
            val p2: Package = getPackage("")
            val metadata: Map<String, Any> = metadata
            val metadataString: String = getMetadataString("key", "default")
            val paywall: PaywallData? = paywall

            Offering(
                identifier = identifier,
                serverDescription = serverDescription,
                metadata = metadata,
                availablePackages = availablePackages,
            )

            Offering(
                identifier = identifier,
                serverDescription = serverDescription,
                metadata = metadata,
                availablePackages = availablePackages,
                paywall = paywall
            )
        }
    }
}
